soccer/gameplay/plays/testing/four_courners_pass.py

```diff
@@ -0,0 +1,229 @@
+		self.defense_speed = variable_square/3.0
+
+		# picks the direction to pass to. TODO make actual smart pass selection
+		self.direction = 1
+
+	# constanly running changes and updates
+	def execute_running(self):
+		# checks if there is an offensive behavior for the various plays and set the normal speed
+		if (self.has_subbehavior_with_name('move1')) :
+			robo = self.subbehavior_with_name('move1').robot
+			if (robo != None) :
+				robo.set_max_speed(self.normal_speed)
+
+		if (self.has_subbehavior_with_name('move2')) :
+			robo = self.subbehavior_with_name('move2').robot
+			if (robo != None) :
+				robo.set_max_speed(self.normal_speed)
+
+		if (self.has_subbehavior_with_name('capture')):
+			robo = self.subbehavior_with_name('capture').robot
+			if (robo != None) :
+				robo.set_max_speed(self.normal_speed)
+
+		if (self.has_subbehavior_with_name('premove')) :
+			robo = self.subbehavior_with_name('premove').robot
+			if (robo != None) :
+				robo.set_max_speed(self.normal_speed)
+
+		# checks if there is a chasing robot and set their speed to defense speed
+		if (self.has_subbehavior_with_name('Chasing')):
+			robo = self.subbehavior_with_name('Chasing').robot
+			if (robo != None) :
+				robo.set_max_speed(self.defense_speed)
+
+		# keep the iso point stuck in the corner
+		if (self.has_subbehavior_with_name('Iso')) :
+			self.iso.pos = robocup.Point(-constants.Field.Width/2, 0)
+
+		# chasing robot positon should always follow the ball within a smaller inside box of the
+		# four corners.
+		if (self.has_subbehavior_with_name('Chasing')) :
+			self.chaser.pos = self.cut_of_pos(main.ball().pos)
+
+		#draw the four courner field
+		self.paint_the_field()
+
+	def on_enter_setup(self):
+		# find where we will consider the ball is closest too and passing from
+		closestPt = min(self.square_points, key=lambda pt: pt.dist_to(main.ball().pos))
+
+		#points that the other robots can move to
+		self.otherPts = list(self.square_points)
+		self.otherPts.remove(closestPt)
+
+		# remove the point that's diagonal of the closest point
+		farthestPt = max(self.otherPts,
+						key=lambda pt: pt.dist_to(main.ball().pos))
+
+		self.otherPts.remove(farthestPt)
+
+
+		# move the other two robots to the other point locations
+		self.add_subbehavior(skills.move.Move(self.otherPts[0]), 'move1', required = False)
+		self.add_subbehavior(skills.move.Move(self.otherPts[1]), 'move2', required = False)
+		# send the closest robot to capture the ball
+		self.add_subbehavior(skills.capture.Capture(), 'capture')
+
+	def on_exit_setup(self):
+		self.remove_subbehavior('capture')
+
+	def on_enter_passing(self):
+		# get the point passing from
+		kickFrom = min(self.square_points,
+					   key=lambda pt: pt.dist_to(main.ball().pos))
+		# get the index of where you're passing from
+		kickFromIdx = self.square_points.index(kickFrom)
+		# decide which direction you're passing to
+		self.safer_pass()
```

Comment:
This feels weird to have safer pass set a global variable when it's not being used anywhere else. I think it would be better to return which one